src/core/models.py

```diff
@@ -565,6 +566,13 @@ def snapshot_self(self, article, force_update=True):
                 setattr(frozen_author, k, v)
             frozen_author.save()
 
+        # now update the CRediT records
+        credit_records = CreditRecord.objects.filter(article=article, author=self)
+
+        for credit_record in credit_records:
+            credit_record.frozen_author = frozen_author
+            credit_record.save()
+
```

Comment:
OK, so I wasn't sure what force_update meant in "if frozen_author and force_update", and assumed it was a unique case, but that we would always want to save the credit_records when we freeze an article. Is that not right?

Reply:
Yes, I agree it's a good idea to update credit records in this spot, it just needs to be inside the first `if` clause / indented four spaces.

WRT to `force_update`, it is defined on `snapshot_authors` if you search for it (granted, it should be defined on `snapshot_self`). It is "Whether or not to update existing records". So this `if` block is supposed to run when there are existing frozen authors and when they need to be updated, which is a lot of the time, like in the pre-publication stage, IIRC.

With the current indentation, you are changing the code below from `if / else` to [`for / else`](https://book.pythontips.com/en/latest/for_-_else.html#else-clause), which is not what we want, because the `else` will always run. We only want lines 577 etc. to run if the conditions on line 564 are not met.